sympy/core/expr.py

```diff
@@ -1940,6 +1940,21 @@ def extract_multiplicatively(self, c):
            x/6
 
         """
+        if (type(c) == int and c < 0):
+            return Expr.extract_multiplicatively(self, c*S.ImaginaryUnit*-S.ImaginaryUnit)
```

Comment:
I don't understand why the ImaginaryUnit is being used. Overall, I prefer the approach at #12270 . Is there something that your modifications do that the other PR does not?